pumpkin-lib/src/engine/cp/propagation/propagator_constructor_context.rs

```diff
@@ -66,6 +67,37 @@ impl PropagatorConstructorContext<'_> {
         var
     }
 
+    /// Subscribes the propagator to the given [`DomainEvents`] when they are undone during
+    /// backtracking.
+    ///
+    /// The domain events determine when [`Propagator::notify_backtrack()`] will be called on the
+    /// propagator. The [`LocalId`] is internal information related to the propagator,
+    /// which is used when calling [`Propagator::notify_backtrack()`] to identify the variable.
+    ///
+    /// Each variable *must* have a unique [`LocalId`]. Most often this would be its index of the
+    /// variable in the internal array of variables.
+    ///
+    /// Note that the [`LocalId`] is used since internally the propagator variable is a wrapper
+    /// around a variable 'view'.
+    pub fn register_for_backtrack_events<Var: IntegerVariable>(
```

Comment:
> Does this need to be called in addition to register? Or instead of register? How do these LocalIds relate to the other registered ones

This should be called in addition to register if necessary; it can also be called separately from `register` if desired. These LocalIds should be the same as the ones for the other registered one but this is not necessary

Reply:
Can we clarify that in the documentation? Because I can see this being confusing.

Reply:
Done, please let me know if this is clear enough